File: apps/backend/src/EnterpriseAgentOs.Domain/Interfaces/IAgentLogService.cs
namespace EnterpriseAgentOs.Domain.Interfaces;

public interface IAgentLogService
{
    Task<List<AgentLogRecord>> ListForAgentAsync(Guid agentId, DateTime? before, int limit, CancellationToken ct = default);
    Task<GlobalLogsPage> ListGlobalAsync(GlobalLogFiltersInput filters, CancellationToken ct = default);
    Task<AgentLogRecord> AppendAsync(AgentLogRecord record, CancellationToken ct = default);
    Task<AgentLogRecord> SendMessageAsync(Guid agentId, string content, Guid userId, CancellationToken ct = default);

    // Audit (merged from Entities/Audit)
    Task RecordToolCallAsync(Guid agentId, Guid? userId, string skillName, string action,
        string paramsJson, string? resultSummary, long durationMs, CancellationToken ct = default);
    Task<(List<AgentLogRecord> Items, int Total)> GetAuditLogAsync(Guid agentId, int limit, int offset, CancellationToken ct = default);
    Task<Dictionary<string, AgentLogRecord>> GetResultsByCorrelationAsync(Guid agentId, IReadOnlyCollection<string> correlationIds, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/analytics/IAgentLogService.java
package com.enterpriseagentos.backendjava.domain.features.analytics;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public interface IAgentLogService {
    CompletableFuture<List<AgentLogRecord>> listForAgentAsync(UUID agentId, Instant before, int limit);
    CompletableFuture<GlobalLogsPage> listGlobalAsync(GlobalLogFiltersInput filters);
    CompletableFuture<AgentLogRecord> appendAsync(AgentLogRecord record);
    CompletableFuture<AgentLogRecord> sendMessageAsync(UUID agentId, String content, UUID userId);
    CompletableFuture<Map<String, AgentLogRecord>> getResultsByCorrelationAsync(UUID agentId, Collection<String> correlationIds);
}
